app/controllers/admin/base_controller.rb

```diff
@@ -1,12 +1,12 @@
 class Admin::BaseController < ApplicationController
-  # before_action :require_admin
+  before_action :require_admin
 
   def dashboard
   end
 
   private
 
-  # def require_admin
-  #   redirect_to root_path unless current_user && current_user.admin?
-  # end
+  def require_admin
+    render file: 'public/404.html' unless current_admin? 
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.<br>Trailing whitespace detected.